geopandas/io/file.py

```diff
@@ -104,8 +104,7 @@ def to_file(df, filename, driver="ESRI Shapefile", schema=None,
     with fiona.drivers():
         with fiona.open(filename, 'w', driver=driver, crs=df.crs,
                         schema=schema, **kwargs) as colxn:
-            for feature in df.iterfeatures():
-                colxn.write(feature)
+            colxn.writerecords((i for i in df.iterfeatures()))
```

Comment:
I think you don't need to do the generator comprehension, but can just return the result from `iterfeatures` ? (so `writerecords(df.iterfeatures())` ?

Reply:
maybe. Will try.

Reply:
@jorisvandenbossche I just made a generator to save a bit of memory. I'm just worried that 1M records that I make sometimes may consume significant amount of RAM.

Reply:
the output of `df.iterfeatures` is already a generator:

```
In [88]: df = GeoDataFrame({'geom': [Point(x, x) for x in range(100)],
    ...:                    'value1': np.arange(100, dtype='int64'),
    ...:                    'value2': np.array([1, 2, 1, 4]*25, dtype='int64')}, geometry='geom')

In [89]: df.iterfeatures()
Out[89]: <generator object GeoDataFrame.iterfeatures at 0x7f009c4ff9e8>
```

Reply:
ah, all right, will fix